micro-rdk/src/common/system.rs

```diff
@@ -0,0 +1,91 @@
+use std::sync::{Arc, LazyLock};
+use std::time::Duration;
+
+#[cfg(not(feature = "esp32"))]
+use async_io::Timer;
+use async_lock::Mutex as AsyncMutex;
+
+use super::app_client::{AppClient, PeriodicAppClientTask};
+use super::log::LogUploadTask;
+use super::runtime::terminate;
+
+pub(crate) static SHUTDOWN_EVENT: LazyLock<Arc<AsyncMutex<Option<SystemEvent>>>> =
+    LazyLock::new(|| Arc::new(AsyncMutex::new(None)));
+
+pub(crate) enum SystemEvent {
+    Restart,
+    #[allow(dead_code)]
+    DeepSleep(Option<Duration>),
+}
+
+impl SystemEvent {
+    fn action_string(&self) -> String {
+        match self {
+            Self::Restart => "restart".to_string(),
+            Self::DeepSleep(dur) => dur.as_ref().map_or("enter deep sleep".to_string(), |d| {
+                format!("enter deep sleep for {} microseconds", d.as_micros())
```

Comment:
how would you go about configuring other wake up sources?

Reply:
was that part of the scope of this? If so I can add that in a follow-up PR where I change the `DeepSleep` enum variant to be a struct with all of the possible settings. Then you would configure these settings through the agent config (just as you would the active timeout)